### DISCUSSION TITLE
High resolution screenshot (low-priority)

### DISCUSSION THREAD

**Discussion by oliviertassinari** [argos-ci/argos#1141] [Ideas]
Right now, Argos CI is configured to handle images that are 1 device px = 1 CSS pixel. I wonder if it would make sense in the future to support higher-resolution screenshots.

I can't find any screenshot tool supporting this, but I suspect that it's trending toward becoming the industry norm (meaning we will start to see budget Windows option support this in the next 5 years).

**Comment by gregberge** [comment]
Hey @oliviertassinari, I think you can already do it using Playwright (https://playwright.dev/docs/api/class-browser#browser-new-context-option-device-scale-factor).

**Comment by oliviertassinari** [comment]
Yes, but once the screenshot is displayed in the UI of Argos, it's displayed too big.

**Comment by gregberge** [comment]
@oliviertassinari still needed?

**Reply by oliviertassinari** [reply]
We haven't started to take x2 image screenshots because it would likely slow down by x4 the CI (x4 more pixels).

But it sounds to me that yes, this is eventually the direction to take (maybe only really in demand in 5 years?). For the scale of the images to be shown 1:1 in the UI of Argos, the scale information needs to be communicated, somehow.

**Comment by gregberge** [comment]
Screenshots are now high resolutions everywhere.

**Reply by oliviertassinari** [reply]
To be clear, this was about:

1. Extracting from the metadata of the image the devicePixelRatio
2. Showing the image in the UI with the correct dimension

**Reply by gregberge** [reply]
Ok I see, it's only relevant for small component images.
